mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import warnings
+
+from mmcv.runner import HOOKS
+from mmcv.runner.dist_utils import master_only
+from mmcv.runner.hooks.checkpoint import CheckpointHook
+from mmcv.runner.hooks.logger.wandb import WandbLoggerHook
+
+from mmdet.core import DistEvalHook, EvalHook
+
+
+@HOOKS.register_module()
+class WandbLogger(WandbLoggerHook):
+    """WandbLogger logs metrics, saves model checkpoints as W&B Artifact, and
```

Comment:
After consideration, we can named this hook `MMDetWandbHook`. It's better to end with `Hook` for clarity of its function.